### ISSUE TITLE
Float view doen't autoclose on opening a file

### ISSUE THREAD

**Issue by i-d-lytvynenko** [nvim-tree/nvim-tree.lua#2624]
### Description

When hitting enter with the cursor on a line with some file name, the file opens in a new buffer, but the nvim-tree doesn't autoclose. Upon trying to move cursor in the nvim-tree window, a number of errors pop up. For some reason the problem does not occur with ```hijack_cursor = false```. The last commit without this bug is 96a783fbd606a458bcce2ef8041240a8b94510ce.

### Neovim version

```text
NVIM v0.9.2
Build type: RelWithDebInfo
LuaJIT 2.1.1694082368
```

### Operating system and version

Windows 10

### nvim-tree version

b8c3a23

### Clean room replication

```lua
vim.g.loaded_netrw = 1
vim.g.loaded_netrwPlugin = 1

vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=F:/nvt-min/site]]
local package_root = "F:/nvt-min/site/pack"
local install_path = package_root .. "/packer/start/packer.nvim"
local function load_plugins()
  require("packer").startup {
    {
      "wbthomason/packer.nvim",
      "nvim-tree/nvim-tree.lua",
      "nvim-tree/nvim-web-devicons",
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. "/plugin/packer_compiled.lua",
      display = { non_interactive = true },
    },
  }
end
if vim.fn.isdirectory(install_path) == 0 then
  print "Installing nvim-tree and dependencies."
  vim.fn.system { "git", "clone", "--depth=1", "https://github.com/wbthomason/packer.nvim", install_path }
end
load_plugins()
require("packer").sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua setup()]]
vim.opt.termguicolors = true
vim.opt.cursorline = true

_G.setup = function()
  require("nvim-tree").setup {
    hijack_cursor = true,
    view = {
      float = {
        enable = true,
      },
    },
  }
end
```

### Steps to reproduce

1. ```nvim -nu F:/nvt-min/nvt-min.lua```
2. ```:e .```
3. Position cursor on some file
4. Hit Enter

**Comment by alex-courtis** [comment]
Confirmed.

`:e .` is opening the tree in the current window and the issue can be reproduced.

Does not occur when opening the tree as a floating window.

Tree does not open when `hijack_directories.enable = false`

**Comment by alex-courtis** [comment]
Should `:e .` open a floating window when hijacking? Documentation is unclear however it intuitively should.

The error on open file should be fixed:
```
Error detected while processing CursorMoved Autocommands for "NvimTree_*":
Error executing lua callback: ...-dev/site/pack/packer/start/master/lua/nvim-tree/lib.lua:27: Invalid window id: 1000
stack traceback:
        [C]: in function 'nvim_win_get_cursor'
        ...-dev/site/pack/packer/start/master/lua/nvim-tree/lib.lua:27: in function 'get_node_at_cursor'
        .../nvt-dev/site/pack/packer/start/master/lua/nvim-tree.lua:121: in function 'place_cursor_on_node'
        .../nvt-dev/site/pack/packer/start/master/lua/nvim-tree.lua:238: in function <.../nvt-dev/site/pack/packer/start/master/lua/nvim-tree.lua:236>
```

**Comment by alex-courtis** [comment]
`view.get_winnr()` is incorrect here. The window it returns 1000 no longer exists.

https://github.com/nvim-tree/nvim-tree.lua/blob/dc839a72a6496ce22ebd3dd959115cf97c1b20a0/lua/nvim-tree/lib.lua#L22

**Comment by alex-courtis** [comment]
@i-d-lytvynenko I would be most grateful if you tested a fix #2627

```sh
cd /path/to/nvim-tree.lua
git pull
git checkout 2624-float-hijack-dir
```

When you're finished testing:
```sh
git checkout master
```

**Comment by i-d-lytvynenko** [comment]
I confirm that this solves the problem.
